### PR TITLE
An option to ingest ee asset as a table/featureCollection

### PR DESCRIPTION
**Pull request by deepgabani8**

### CODE REVIEW

File: weather_mv/README.md [hunk 1]
```diff
@@ -323,6 +323,7 @@ _Command options_:
 * `--ee_asset`: (required) The asset folder path in earth engine project where the tiff image files will be pushed.
   It should be in format: `projects/<project-id>/assets/<asset-folder>`. Make sure that <asset-folder> is there
   under <project-id> in earth engine assets. i.e. projects/my-gcp-project/assets/my/foo/bar.
+* `--ee_asset_type`: The type of asset to ingest in the earth engine.
```
Comment (alxmrs) [inline weather_mv/README.md:326]: Please indicate the default value.

File: weather_mv/loader_pipeline/ee.py [hunk 2]
```diff
@@ -230,6 +231,8 @@ def add_parser_arguments(cls, subparser: argparse.ArgumentParser):
         subparser.add_argument('--ee_asset', type=str, required=True, default=None,
                                help='The asset folder path in earth engine project where the tiff image files'
                                ' will be pushed.')
+        subparser.add_argument('--ee_asset_type', type=str, default='IMAGE',
```
Comment (alxmrs) [inline weather_mv/loader_pipeline/ee.py:234]: Let's use choices instead of a string. 

https://docs.python.org/3/library/argparse.html#choices

**Review by alxmrs — COMMENTED** [review]
Do feature collections consume COGs? Does this work e2e (can I see the output)?

**Comment by deepgabani8** [comment]
@alxmrs 
I missed that point. Feature collection wouldn't be able to consume COGs.
We shouldn't merge this here then, right?

**Comment by alxmrs** [comment]
No, let's not merge this. I'd like to see the results of e2e prototypes before we land this change.

File: weather_mv/loader_pipeline/ee.py [hunk 3]
```diff
@@ -478,33 +507,49 @@ def start_ingestion(self, asset_request: t.Dict) -> str:
...
-        start_time = tiff_data.start_time
-        end_time = tiff_data.end_time
-        properties = tiff_data.properties
+    def process(self, asset_data: AssetData) -> t.Iterator[str]:
+        """Uploads an asset into the earth engine."""
+        target_path = asset_data.target_path
+        asset_name = os.path.join(self.ee_asset, asset_data.name)
+        start_time = asset_data.start_time
+        end_time = asset_data.end_time
+        properties = asset_data.properties
```
Comment (alxmrs) [inline weather_mv/loader_pipeline/ee.py:527]: Optional style nit: My personal preference is that extracting out these variables is unnecessary; you can just access them from `asset_data`.

**Review by alxmrs — APPROVED** [review]
LGTM. One ask: Please file a bug s.t. we can scale EE ingestion to large files (which are bottlenecked on memory).

File: weather_mv/loader_pipeline/ee.py [hunk 4]
```diff
@@ -392,56 +401,74 @@ def process(self, uri: str) -> t.Iterator[TiffData]:
...
+                    # Copy in-memory tiff to gcs.
+                    target_path = os.path.join(self.asset_location, file_name)
+                    with FileSystems().create(target_path) as dst:
+                        shutil.copyfileobj(memfile, dst, WRITE_CHUNK_SIZE)
+            # For feature collection ingestions.
+            elif self.ee_asset_type == 'TABLE':
+                file_name = f'{asset_name}.csv'
+
+                df = xr.Dataset.to_dataframe(ds)
+                df = df.reset_index()
```
[the remark this answers is not in the document]
Comment (deepgabani8) [reply weather_mv/loader_pipeline/ee.py:443]: Yes. Added [this bug](https://github.com/google/weather-tools/issues/228).

**Pull request merged**